pkg/controllers/localbuild/argo.go

```diff
@@ -53,7 +68,74 @@ func (r *LocalbuildReconciler) ReconcileArgo(ctx context.Context, req ctrl.Reque
 	if result, err := argocd.Install(ctx, resource, r.Client, r.Scheme, r.Config); err != nil {
 		return result, err
 	}
-
 	resource.Status.ArgoCD.Available = true
+
+	// Let's patch the existing argocd admin secret if devmode is enabled to set the default password
+	if r.Config.DevMode {
+		// Hash password using bcrypt
+		password := argocdDevModePassword
+		hashedPassword, err := bcrypt.GenerateFromPassword([]byte(password), 0)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("Error hashing password: %w", err)
+		}
+
+		// Prepare the patch for the Secret's `stringData` field
+		patchData := map[string]interface{}{
+			"stringData": map[string]string{
+				"accounts.developer.password":      string(hashedPassword),
+				"accounts.developer.passwordMtime": time.Now().Format(time.RFC3339),
+			},
+		}
+		// Convert patch data to JSON
+		patchBytes, err := json.Marshal(patchData)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("Error marshalling patch data: %w", err)
+		}
+
+		kubeClient, err := k8s.GetKubeClient()
```

Comment:
The reconciler has client already. You can just use that instead. e.g. `r.Client.`. You can also change passwords through ArgoCD api.

Reply:
> You can also change passwords through ArgoCD api.

That will require to be done in a 2nd step while here the idea is to have the account developer/developer available when packages are installed too

Reply:
> > You can also change passwords through ArgoCD api.
> 
> That will require to be done in a 2nd step while here the idea is to have the account developer/developer available when packages are installed too

Yeah I am starting to think we should set them through API for both Gitea and ArgoCD in every invocation of idpbuilder. Instead of relying on manifests.